dig_test.go

```diff
@@ -919,6 +919,48 @@ func TestGroups(t *testing.T) {
 		}), "invoke failed")
 	})
 
+	t.Run("groups are provided via option", func(t *testing.T) {
```

Comment:
Can we add a test case for the same function producing multiple values to the
same group?

```
c.Provide(func() (int, int, string) { ... }, Group("foo"))
```

The consumer will expect a `[]int` and `[]string`, both with group "foo".